src/main/webapp/view/transcript/transcript.js

```diff
@@ -113,60 +85,106 @@ function seekTranscript(currentTime) {
   if (currentTimeMs < currentTranscriptLine.startTimestampMs) {
     return;
   }
-  if (isWithinCurrentTimeRange(currentTimeMs)) {
-    addBold(currentTranscriptLine);
+  if (currentTranscriptLine.isWithinCurrentTimeRange(currentTimeMs)) {
+    currentTranscriptLine.addBold();
     return;
   }
-  removeBold(currentTranscriptLine);
+  currentTranscriptLine.removeBold();
   currentTranscriptLine = currentTranscriptLine.nextElementSibling;
-  scrollToTopOfTranscript(currentTranscriptLine);
-  addBold(currentTranscriptLine);
+  currentTranscriptLine.scrollToTopOfTranscript();
+  currentTranscriptLine.addBold();
   // TODO: Handle the case where the video isn't only playing.
 }
 
 /**
- * Bolds the text in `transcriptLineLiElement` if it is not already
- * bolded.
+ * Creates a transcript line element containing the text,
+ * start time, and end time from `transcriptLine`.
  */
-function addBold(transcriptLineLiElement) {
-  if (isBolded(transcriptLineLiElement)) {
-    return;
+class TranscriptLine extends HTMLElement {
+  /**
+   * Creates a custom HTML element representing a transcript line.
+   *
+   * <p>Uses the template and slots defined in `TRANSCRIPT_TEMPLATE` to
+   * help create the transcript line.
+   *
+   * @param transcriptLine The transcriptLine from `ENDPOINT_TRANSCRIPT`
+   *     whose `attributes` should be used.
+   */
+  constructor(transcriptLine) {
+    super();
```

Comment:
It is best practise to not do work in the constructor. Please see: https://docs.google.com/document/d/14CEPF-d3eF6HzG4AhviQiRbdEaVF8_BDOwKHMUmqFEQ/edit

I would use a factory method to create the object.

Reply:
This is the same situation as in the DiscussionComment class, which we talked about and okay-ed.  Both use templates and need to initialize various things before they are rendered.

Reply:
I ended up talking to my co-worker about this and independent of the language, doing work in the constructor is not a good practise. It is best to create a static method in the class which does all the work and passes the fields to the constructor.

Reply:
The part I have a problem with in this ```createTimestampRange()``` That seems to me the part that is doing some work. If we can remove that and put it in some kind of initialization method that would work for me.

Reply:
(Resolved in chat) I kept the code for attaching the shadow in the constructor but moved the code for creating the timerange and updating the slots to a static method.